datafusion/core/tests/sql/subqueries.rs

```diff
@@ -0,0 +1,461 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use super::*;
+use crate::sql::execute_to_batches;
+use datafusion::assert_batches_eq;
+use datafusion::prelude::SessionContext;
+use log::debug;
+
+#[cfg(test)]
+#[ctor::ctor]
+fn init() {
```

Comment:
TIL `ctor`. That is quite cool 👍

Reply:
Logging has been a challenge:
1. need to add `--nocapture` to tests to see log statements
2. need to add `RUST_LOG=debug` environment variable
3. need to init the logger

This prevented me from accidentally checking in `println!()` statements.